server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java

```diff
@@ -382,16 +384,37 @@ synchronized void stopWatcher() {
         }
     }
 
-    private WatchKey enableSettingsWatcher(WatchKey previousKey, Path settingsDir) throws IOException {
+    // package private for testing
+    long retryDelayMillis(int failedCount) {
+        assert failedCount < 31;
+        return 10 * (1 << failedCount) + Randomness.get().nextInt(10);
```

Comment:
What's the randomness for here?

Reply:
For jitter, so that two processes launched at the same time don't accidentally overlap.

Reply:
Ah, so we could have the same files being read by two different ES instances? Fair.

Reply:
Maybe max timeout something like 3-5 seconds, I wouldn't want more than that in case it blocked the thread too much